hathor/transaction/block.py

```diff
@@ -93,6 +93,26 @@ def create_from_struct(cls, struct_bytes: bytes, storage: Optional['TransactionS
 
         return blc
 
+    def set_height(self, height: int) -> None:
```

Comment:
In this case, shouldn't we store the expected height in a separate metadata? I mean, we trust `meta.height` as something calculated from the parent and which would never be wrong. It seems we are mixing two different heights: expected and actual.

Reply:
Maybe, moving it to a different metadata just pushes that problem to when the height needs to be used, this was added because the actual height needed to be used before we could actually be able to calculate it. I'm not entirely sure if this is still the case.

I think the need for this can be revisited after the sync is functional again. It should be simple to just remove it and see if everything is still working.

Reply:
Outdated conversation.